File: algorithms/python/binary_search.py
def binary_search(lst, item):
    low, high = 0, len(lst) - 1
    while low <= high:
        mid = (low + high) // 2
        guess = lst[mid]
        if guess == item:
            return mid
        elif guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None


if __name__ == '__main__':
    lst = range(0, 101, 2)
    print(binary_search(lst, 32))
    print(binary_search(lst, 31))

File: algorithms/C/binary_search.c
#include <stdio.h>
#include <stdlib.h>

int binary_search(int[], int, int);

int main()
{
  int list[51];
  for (int i = 0; i <= 50; i++)
  {
    list[i] = i * 2;
  }
  printf("%d\n", binary_search(list, 51, 32));
  printf("%d\n", binary_search(list, 51, 31));
}

int binary_search(int list[], int n, int item)
{
  int low = 0, high = n - 1;
  while (low <= high)
  {
    int mid = (low + high) / 2;
    int guess = list[mid];
    if (guess == item)
      return mid;
    else if (guess > item)
      high = mid - 1;
    else
      low = mid + 1;
  }
  return -1;
}